src/mpl/include/mpl_gpu.h

```diff
@@ -21,7 +21,12 @@ typedef enum {
     MPL_GPU_POINTER_MANAGED
 } MPL_pointer_type_t;
 
-int MPL_gpu_query_pointer_type(const void *ptr, MPL_pointer_type_t * attr);
+typedef struct {
+    MPL_pointer_type_t type;
+    MPL_gpu_dev_id_t device;
```

Comment:
Do we need this to be opaque?  It can't be used as an array index, for example, if it's opaque.  Device ID can always be an integer.  Device handles can be hidden inside MPL.

Reply:
Sure I can change device id to integer.

Reply:
this is fixed